test/basicfuns.jl

```diff
@@ -137,6 +138,11 @@ end
     @test isnan(logsumexp([NaN, 9.0]))
     @test isnan(logsumexp([NaN, Inf]))
     @test isnan(logsumexp([NaN, -Inf]))
+
+    # issue #63
+    a = logsumexp(i for i in range(-500, stop = 10, length = 1000) if true)
+    b = logsumexp(range(-500, stop = 10, length = 1000))
+    @test a == b
```

Comment:
Even without the `if` you get a `Generator`, which is enough to test the method:
```suggestion
    # logsumexp with general iterables (issue #63)
    x = logsumexp(range(-500, stop = 10, length = 1000))
    @test logsumexp(i for i in x) == logsumexp(x)
```

Reply:
```suggestion
    # logsumexp with general iterables (issue #63)
    xs = range(-500, stop = 10, length = 1000)
    @test logsumexp(x for x in xs) == logsumexp(xs)
```